cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -625,3 +637,52 @@ function generateProductUrl({ objectID, productUrl, queryID, indexName }) {
     url.searchParams.append('indexName', indexName);
     return url.href;
 }
+
+/**
+ * Calculate the display price for an item when lazy loading pricing is disabled
+ * @param {Object} item The item object
+ * @return {number} The calculated sales price
+ */
+function calculateDisplayPrice(item) {
+
+    var promotions;
+    var calloutMsg = '';
+
+    if (algoliaData.recordModel === 'master-level' && item.variants) {
+        promotions = item.variants && item.variants.length > 0 ? item.variants[0].promotions : item.promotions;
+        variant = item.variants && item.variants.length > 0 ? item.variants[0] : item;
+    } else {
+        promotions = item.promotions;
+        variant = item;
+    }
+
+    if (promotions && promotions[algoliaData.currencyCode]) {
+        var productPromos = promotions[algoliaData.currencyCode];
+        var minPrice = algoliaData.recordModel === 'master-level' ? (variant.price[algoliaData.currencyCode] ? variant.price[algoliaData.currencyCode] : variant.price) : item.price;
```

Comment:
It would be good to simplify this condition, it's hard to read.
I see that you have introduced a `item.displayPrice`. Could we rely only on it for display, and stop overriding `item.price` in the transformItem function? It would avoid having to do those checks.

Can be done in another PR if you prefer.

Reply:
I prefer to do it via another PR because I will already implement a lazy load that depends on this PR.